conbench/entities/context.py

```diff
@@ -12,12 +14,34 @@
 )
 
 
+# Maybe EntityMixin should be a generic class
+# https://mypy.readthedocs.io/en/latest/generics.html#defining-subclasses-of-generic-classes
```

Comment:
https://discuss.python.org/t/a-way-to-typehint-a-return-type-in-parent-based-on-return-type-of-a-child/17020